src/Domain/Creators/TimeDependence/UniformRotationAboutZAxis.cpp

```diff
@@ -0,0 +1,148 @@
+  for (size_t i = 1; i < number_of_blocks; ++i) {
+    result[i] = result[0]->get_clone();
+  }
+  return result;
+}
+
+template <size_t MeshDim>
+std::unordered_map<std::string,
+                   std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+UniformRotationAboutZAxis<MeshDim>::functions_of_time() const noexcept {
+  std::unordered_map<std::string,
+                     std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+      result{};
+  // We use a `PiecewisePolynomial` with 2 derivs since some transformations
+  // between different frames for moving meshes can require Hessians.
+  result[function_of_time_name_] =
+      std::make_unique<FunctionsOfTime::PiecewisePolynomial<2>>(
+          initial_time_,
+          std::array<DataVector, 3>{{{0.0}, {angular_velocity_}, {0.0}}});
+  return result;
+}
+
+/// \cond
+template <>
+auto UniformRotationAboutZAxis<1>::map_for_composition() const noexcept
+    -> MapForComposition {
+  return MapForComposition{domain::CoordinateMaps::Identity<1>{}};
+}
+
+template <>
+auto UniformRotationAboutZAxis<2>::map_for_composition() const noexcept
+    -> MapForComposition {
+  return MapForComposition{domain::CoordinateMaps::TimeDependent::Rotation<2>{
+      function_of_time_name_}};
+}
+
+template <>
+auto UniformRotationAboutZAxis<3>::map_for_composition() const noexcept
+    -> MapForComposition {
+  using ProductMap = domain::CoordinateMaps::TimeDependent::ProductOf2Maps<
+      domain::CoordinateMaps::TimeDependent::Rotation<2>,
+      domain::CoordinateMaps::Identity<1>>;
+  return MapForComposition{
+      ProductMap{domain::CoordinateMaps::TimeDependent::Rotation<2>{
+                     function_of_time_name_},
+                 domain::CoordinateMaps::Identity<1>{}}};
+}
+
+template <size_t Dim>
+bool operator==(const UniformRotationAboutZAxis<Dim>& lhs,
+                const UniformRotationAboutZAxis<Dim>& rhs) noexcept {
+  return lhs.initial_time_ == rhs.initial_time_ and
+         lhs.angular_velocity_ == rhs.angular_velocity_ and
+         lhs.function_of_time_name_ == rhs.function_of_time_name_;
+}
+
+template <size_t Dim>
+bool operator!=(const UniformRotationAboutZAxis<Dim>& lhs,
+                const UniformRotationAboutZAxis<Dim>& rhs) noexcept {
+  return not(lhs == rhs);
+}
+
+#define GET_DIM(data) BOOST_PP_TUPLE_ELEM(0, data)
+
+#define INSTANTIATION(r, data)                                   \
+  template class UniformRotationAboutZAxis<GET_DIM(data)>;       \
+  template bool operator==<GET_DIM(data)>(                       \
+      const UniformRotationAboutZAxis<GET_DIM(data)>&,           \
+      const UniformRotationAboutZAxis<GET_DIM(data)>&) noexcept; \
+  template bool operator!=<GET_DIM(data)>(                       \
+      const UniformRotationAboutZAxis<GET_DIM(data)>&,           \
+      const UniformRotationAboutZAxis<GET_DIM(data)>&) noexcept;
+
+GENERATE_INSTANTIATIONS(INSTANTIATION, (1, 2, 3))
```

Comment:
is the class defined and the identity in 1D, or undefined and should static-assert dim > 1? If the latter, remove the 1D instantiation here

Reply:
Good catch! I meant to remove the 1D case but missed the instantiation here